### PR TITLE
append_without_duplicates method added

### PR DESCRIPTION
**Pull request by souvik-databricks**
Completes #4 

Local test code execution result below:
image

### CODE REVIEW

File: tests/test_public_interface.py [hunk 1]
```diff
@@ -388,3 +388,39 @@ def test_copy_delta_table(tmp_path):
...
+        (1,"A","B"),
+        (2,"R","T"),
+        (3,"X","Y")
+    ]
+    df = spark.createDataFrame(data, ["col1","col2","col3"])
+    df.write.format("delta").save(path)
+
+    deltaTable = DeltaTable.forPath(spark, path)
+
+    append_data = spark.createDataFrame(
```
Comment (MrPowers) [inline tests/test_public_interface.py:406]: This test data doesn't have any data that's already in the Delta table.  We need some duplicate data here to demonstrate that the duplicate data isn't appended, only the new data.

Comment (souvik-databricks) [reply tests/test_public_interface.py:406]: @MrPowers Added the multiple column test also. Results below: 
image

**Comment by MrPowers** [comment]
@robertkossendey @nkarpov - can you please take a look if you have a sec?

**Review by robertkossendey — COMMENTED** [review]
LGTM

**Comment by MrPowers** [comment]
@souvik-databricks - thanks for adding this 🚀

**Pull request merged**